### PR TITLE
Implement accent-insensitive search services and endpoints for organizers, speakers, and places #710

### LINKED ISSUE #710: Story 2.1-BE: Autocomplete Fuzzy Search Services & Endpoints
**As a** backend developer,  
**I want** to implement high-speed fuzzy search services and read endpoints utilizing `unaccent` normalization for organizers, speakers, and places,  
**So that** frontend selectors can retrieve existing matching records in under 300ms to prevent duplicate database entries.

**Acceptance Criteria:**
* **Given** search queries with missing/incorrect accents or capitalization (e.g., "feria libro" or "blaa")
  * **When** evaluated by search services `search_organizers(q)` and `search_speakers(q)` in `events/services/entity_search.py`, and `search_places(q)` in `places/services/place_search.py`
  * **Then** they utilize PostgreSQL `unaccent__icontains` to retrieve accent-insensitive normalized matches (e.g. returning "Biblioteca Luis Ángel Arango").
* **Given** the autocomplete endpoints:
  * `GET /events/api/v1/organizers/search/?q=`
  * `GET /events/api/v1/speakers/search/?q=`
  * `GET /places/api/v1/places/search/?q=`
  * **When** queried with a parameter `q` of at least 2 characters
  * **Then** they return HTTP 200 with `{ "results": [{ "id": 1, "name": "..." }] }`
  * **And** search execution completes in <= 300ms at the 95th percentile (NFR1).

### PR DESCRIPTION
**Pull request by cansadadeserfeliz**
## Summary by CodeRabbit

* **New Features**
  * Added authenticated autocomplete search endpoints for organizers, speakers, and places.
  * Accent-insensitive substring matching (e.g., “jose garcia” matches “José García”).
  * Query handling: absent/empty `q` returns the first results (ordered by name); `q` with 1 character returns an empty list.
  * Organizer and speaker results include `image_url`; place results include `id` and `name` only.
* **Tests**
  * Added unit and API tests covering authentication, query edge-cases, ordering, `limit` enforcement, and response shape.

### CODE REVIEW

File: events/services/organizer_search.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+from events.models import Organizer
+
+
+def search_organizers(q: str, limit: int = 10) -> QuerySet:
+    if len(q) < 2:
+        return Organizer.objects.none()
+    return (
+        Organizer.objects.filter(name__unaccent__icontains=q).order_by('name')[:limit]
+    )
```
[the remark this answers is not in the document]
Comment (cansadadeserfeliz) [reply events/services/organizer_search.py:25]: SearchVector/SearchQuery tokenizes text into lexemes (complete words or stems). When a user types "Tea" and you pass it to SearchQuery("Tea"), PostgreSQL FTS will not match "Teatro
  Mayor" — it expects a complete word token, not a prefix.

File: events/services/speaker_search.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from django.db.models import QuerySet
+
+from events.models import Speaker
+
+
+def search_speakers(q: str, limit: int = 10) -> QuerySet:
+    if len(q) < 2:
+        return Speaker.objects.none()
+    return Speaker.objects.filter(name__unaccent__icontains=q).order_by('name')[:limit]
```
[the remark this answers is not in the document]
Comment (cansadadeserfeliz) [reply events/services/speaker_search.py:23]: Speakers contain first and last names, not dictionary words; do you still recommend to use SearchVector/SearchQuery?

File: events/tests/services/test_entity_search.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    results = list(search_organizers('feria internacional'))
+    assert len(results) == 1
+
+
+@pytest.mark.django_db
+def test_search_organizers_empty_when_q_less_than_2_chars():
+    OrganizerFactory(name='Feria Internacional del Libro')
+    assert list(search_organizers('')) == []
+    assert list(search_organizers('f')) == []
+
```
[the remark this answers is not in the document]
Comment (cansadadeserfeliz) [reply events/tests/services/test_entity_search.py:33]: Same comment as for speakers and organizers: place is usually a proper noun (BLAA, Banco de la República, Tornamesa, etc)
